src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1704,42 +1726,55 @@ def try_create_application_insights(cmd, resource_group, name, location):
                    'You can visit %s/#resource%s/overview to view your '
                    'Application Insights component', appinsights.name, portal_url, appinsights.id)
 
-    return appinsights.instrumentation_key
+    return appinsights
 
 
-def app_insights_update(cmd, client, resource_group, name, app_insights_key=None, app_insights=None, sampling_rate=None, disable=None, no_wait=False):
+def app_insights_update(cmd, client, resource_group, name,
+                        app_insights_key=None, app_insights=None, sampling_rate=None,
+                        disable=None, no_wait=False):
+    """
+    :param app_insights_key: Connection string or Instrumentation key
+    :param sampling_rate: float from 0.0 to 100.0, both included
+    """
     if disable:
-        trace_properties = models_20201101preview.MonitoringSettingProperties(trace_enabled=False)
+        monitoring_setting_properties = models_20201101preview.MonitoringSettingProperties(trace_enabled=False)
     else:
-        trace_properties = client.monitoring_settings.get(resource_group, name).properties
-        if not trace_properties.app_insights_instrumentation_key and not app_insights_key and not app_insights and sampling_rate:
-            CLIError("Can't set '--sampling-rate' without connecting to Application Insights. Please provide '--app-insights' or '--app-insights-key'.")
+        monitoring_setting_properties = client.monitoring_settings.get(resource_group, name).properties
+        if not monitoring_setting_properties.app_insights_instrumentation_key \
+                and not app_insights_key \
+                and not app_insights:
+            CLIError("Can't update application insights without connecting to Application Insights. "
+                     "Please provide '--app-insights' or '--app-insights-key'.")
```

Comment:
Please use a specific error type instead of `CLIError`

Reply:
Sure